worlds/_manual/Rules.py

```diff
@@ -80,19 +80,32 @@ def checkRequireStringForArea(state, area):
 
             if len(item_parts) > 1:
                 item_name = item_parts[0]
-                item_count = int(item_parts[1])
+                item_count = item_parts[1]
 
             total = 0
 
             if require_type == 'category':
-                category_items = [item["name"] for item in base.item_name_to_item.values() if "category" in item and item_name in item["category"]]
+                category_items = [item for item in base.item_name_to_item.values() if "category" in item and item_name in item["category"]]
+                if item_count == 'all':
+                    item_count = sum([base.item_name_to_item[category_item["name"]]['count'] for category_item in category_items])
+                elif item_count == 'half':
+                    item_count = sum([base.item_name_to_item[category_item["name"]]['count'] for category_item in category_items]) / 2
+                else:
+                    item_count = int(item_count)
 
                 for category_item in category_items:
-                    total += state.count(category_item, player)
+                    total += state.count(category_item["name"], player)
 
                     if total >= item_count:
                         requires_list = requires_list.replace(item_base, "1")
             elif require_type == 'item':
+                if item_count == 'all':
+                    item_count = base.item_name_to_item[item_name]['count']
+                elif item_count == 'half':
+                    item_count = base.item_name_to_item[item_name]['count'] / 2
```

Comment:
Boolean logic allows either upper or lower casing of OR/AND, and I think we should do the same here with ALL/HALF. So probably want a lower() or something here.

(Looks like int() still rounds down, so that's perfect. 😄 )